### PR TITLE
Added CHANGELOG.md for release [0.3.0]

### PR DESCRIPTION
**Pull request by Mifrill**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,101 @@
...
+
+    - `#match?` compare message by passed regexp
+
+    - description of `Dialog` feature in README
+
+- `Ferrum::Page::Event` extend of `Concurrent::Event` with implementation of `reset/wait` fix
+ 
+    - implement `Ferrum::Page::Event#iteration` to reuse `synchronize` block on `@iteration` value of `Concurrent::Event`
+
+    - redefinition of `Concurrent::Event#reset` - increase `@iteration` outside of `if @set` block
```
Comment (Mifrill) [inline CHANGELOG.md:41]: @route this fix looks pretty good to make PR to gem `ruby-concurrency`, WDYT?

https://github.com/rubycdp/ferrum/blob/999912928ed8d105723e9edbc9c7d6a71238afb8/lib/ferrum/page.rb#L18-L30

Comment (Mifrill) [reply CHANGELOG.md:41]:
```
    # Reset a previously set event back to the `unset` state
    # with the synchronized increase of iteration counter
    # Has no effect on `set` if the `Event` has not yet been set.
```

Comment (route) [reply CHANGELOG.md:41]: Don't really know if they need it I just wanted to see if iteration has changed but who knows maybe..

**Comment by route** [comment]
Thank you, LGTM!

**Pull request merged**